openwisp_monitoring/check/classes/iperf.py

```diff
@@ -0,0 +1,186 @@
+import json
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist
+from swapper import load_model
+
+from openwisp_controller.connection.settings import UPDATE_STRATEGIES
+
+from .. import settings as app_settings
+from .base import BaseCheck
+
+logger = logging.getLogger(__name__)
+
+Chart = load_model('monitoring', 'Chart')
+Metric = load_model('monitoring', 'Metric')
+Device = load_model('config', 'Device')
+Credentials = load_model('connection', 'Credentials')
+AlertSettings = load_model('monitoring', 'AlertSettings')
+DeviceConnection = load_model('connection', 'DeviceConnection')
+
+
+class Iperf(BaseCheck):
+    def check(self, store=True):
+        try:
+            device = self.related_object
+            device_connection = self._check_device_connection(device)
```

Comment:
Let's only put the code that can raise exception in try-except block. I think only these lines can raise `ObjectDoesNotExist` exception here.

Reply:
Thanks for the suggestion, I've already changed this in my local env, Will push soon :)

Reply:
> Thanks for the suggestion, I've already changed this in my local env, Will push soon :)

Acc to https://github.com/openwisp/openwisp-monitoring/pull/394#discussion_r902404374, `_get_device_connection` method now only return either `device_connection` object or `None` so we can now safely remove `try-except` block from the code.